action.yml

```diff
@@ -36,6 +36,18 @@ runs:
       env:
         YARN_NPM_AUTH_TOKEN: ${{ inputs.npm-token }}
         PUBLISH_NPM_TAG: ${{ inputs.npm-tag }}
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - id: diff
+      shell: bash
+      if: inputs.npm-token == ''
+      run: |
+        LATEST_MAJOR_TAG=$(git describe --abbrev=0 --match "v[0-9]*.[0-9]*.[0-9]*" --tags)
```

Comment:
It looks like this wouldn't work for monorepos. We'll need to loop over each workspace and look for a package-specific tag for each one instead.

Reply:
I'm confused? the test I linked is for the `core` monorepo and it is working appropriately? in this case it correctly got the `v52.0.0` tag... which was the last major tag at the time the job ran.

Reply:
It's running but it's providing the wrong answer. The top-level tags for monorepos don't reflect what will be published, because not all packages are published each time. e.g. if a package was last published in v50, then comparing v52 to v53 would miss a bunch of changes for that package.

Reply:
I see what you're saying now... that makes sense.